src/containers/pages/OrganizationViews/AssignPractitioners/index.tsx

```diff
@@ -0,0 +1,279 @@
+
+/** AssignPractitioner component */
+const AssignPractitioner: React.FC<PropsTypes> = props => {
+  const {
+    serviceClass,
+    fetchOrganizationsCreator,
+    organization,
+    fetchPractitionerRolesCreator,
+    assignedPractitioners,
+  } = props;
+  const [selectedOptions, setSelectedOptions] = useState<OptionsType<SelectOption>>([]);
+
+  useEffect(() => {
+    const organizationId = props.match.params.id;
+    loadOrganization(organizationId, serviceClass, fetchOrganizationsCreator);
+    loadOrgPractitioners(organizationId, serviceClass, fetchPractitionerRolesCreator);
+  }, []);
+
+  if (!organization) {
+    return <Loading />;
+  }
+
+  // TODO - scenario where this request doesn't return all practitioners in a single query
+  /** load all practitioners at least all of those returned
+   * in a single call to Practitioners endpoint
+   * @param {typeof OpenSRPService} service - the OpenSRPService
+   */
+  const loadAllPractitioners = async (service: typeof OpenSRPService = OpenSRPService) => {
+    const serve = new service(OPENSRP_PRACTITIONER_ENDPOINT);
+    const allPractitioners = await serve.list();
+    return formatOptions(allPractitioners);
+  };
+
+  // TODO - Hack in this: typings for the changeHandler function.
+  /** This sets the state selectedOptions
+   * @param {ValueType<SelectOption>} -  the so far selected options
+   * @param {ActionMeta} - information on the change event; custom react-select event
+   */
+  const changeHandler = (chosenOptions: any, { action, removedValue, option }: any) => {
+    if (!chosenOptions) {
+      return;
+    }
+    switch (action) {
+      case 'remove-value':
+      case 'pop-value':
+        if (removedValue.isFixed) {
+          return;
+        } else {
+          const remainingOptions = selectedOptions.filter(o => o.value !== removedValue.value);
+          setSelectedOptions(remainingOptions);
+        }
+        break;
+      case 'select-option':
+        setSelectedOptions([...selectedOptions, option]);
+        break;
+      case 'clear':
+        chosenOptions = chosenOptions.filter((v: any) => !v.isFixed);
+    }
+  };
+
+  // TODO - This will initiate an api request for the same exact data each time someone types
+  /** merges the practitioner records to be shown in the dropdown and returns them as  a promise
+   * @param {string} typedChars - value of select if user types in on it
+   */
+  const promiseOptions = async (typedChars: string) => {
+    // merging practitioners that are assigned to the organization and all practitioners
+    const orgPractitioners = formatOptions(assignedPractitioners, true);
+    const allPractitioners = await loadAllPractitioners();
+    const mergedOptions = {
+      ...keyBy(orgPractitioners, option => option.value),
+      ...keyBy(allPractitioners, option => option.value),
+    };
+    return filterOptions(typedChars, values(mergedOptions));
+  };
```

Comment:
Could we instead make the API call ~1sec after the user stops typing?

Reply:
yeah, i will see how i can factor this in